### ISSUE TITLE
Add marginal density/histogram to geom_hex

### ISSUE THREAD

**Issue by mikejiang** [daattali/ggExtra#27]
It will be nice to able to show geom_hex along with two densities on top and right since there are circumstances where scatterplot is too slow due to the huge number of events.

**Comment by daattali** [comment]
You can do that right now.  Just call `ggMarginal()` with a plot that has a `geom_hex` layer instead of `geom_point`.  

```
library(ggplot2)
library(ggExtra)
d <- ggplot(diamonds, aes(carat, price))
ggMarginal(d + geom_hex())
```

**Comment by hisakatha** [comment]
For your information, I'd like to let @daattali know that `geom_hex` is no longer supported since ggExtra version 0.9, although there is another way to achieve similar plots (http://www.lreding.com/nonstandard_deviations/2017/08/19/cowmarg/).

Example:

(R: 3.6.1, ggplot2: 3.2.1, ggExtra: 0.9)

``` r
library(ggplot2)
library(ggExtra)
p1 <- ggplot(mtcars, aes(wt, mpg))
ggMarginal(p1 + geom_point(), type = "density")
```

![](https://i.imgur.com/OOXspsY.png)

``` r
ggMarginal(p1 + geom_hex(), type = "density")
#> Error: No geom_point layer was found in your scatter plot
ggMarginal(p1 + geom_hex(), type = "histogram")
#> Error: No geom_point layer was found in your scatter plot
```

<sup>Created on 2019-11-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by daattali** [comment]
Thanks. You could technically still do it with ggExtra by also adding `+ geom_point(col="transparent")` but it's not very efficient because the geom_point takes a long time

```
library(ggplot2)
library(ggExtra)
d <- ggplot(diamonds, aes(carat, price))
ggMarginal(d + geom_point(col="transparent")+geom_hex())
```

But I'll reopen the issue since that's not a great solution

**Comment by crew102** [comment]
@daattali , do we want to relax the assumption that there has to be a geom_point layer for ggMarginal to try to create the marginals? In other words, do we want to check if there is either geom_point or geom_hex moving forward, as opposed to requiring there be a geom_point layer?

**Comment by daattali** [comment]
I'm comfortable with that, under the assumption that it will also work correctly of course :) I tried seeing what happens if we check for point OR hex, and didn't change anything else, and the results are incorrect because the marginal density treats the hex as if they are regular points, disregarding the density that each hex point conveys.

```
library(ggplot2)
library(ggExtra)
d <- ggplot(diamonds, aes(carat, price))
```

```
ggMarginal(d + geom_point())
```

![image](https://user-images.githubusercontent.com/952340/69910807-556c0a00-13df-11ea-9537-7b03526bb912.png)


```
ggMarginal(d + geom_hex())
```

![image](https://user-images.githubusercontent.com/952340/69910811-7896b980-13df-11ea-8594-5c5b10b54a5f.png)

**Comment by crew102** [comment]
Ergh, you're right. I'd be possible to use the underlying data, but it would add additional complexity to the codebase and I think not be worth it. I'm partial to simply requiring that a geom_point layer be used.

**Comment by daattali** [comment]
Agreed. This can remain open as I'm not opposed to supporting it if a natural solution presents itself
